sdks/web-ui-sdk/src/main.ts

```diff
@@ -23,54 +46,35 @@ export const flows: BallerineSDKFlows = {
       // Always init with no state. This ensures using init to reset the flow returns to the first step with no data.
       void resetAppState();
 
-      configuration.set(defaultConfiguration);
-
-      const { translations: _translations, ...configWithoutTranslations } = config;
-
-      // Extract config from query params
-      const { clientId, flowName, ...endUserInfoFromQueryParams } = getConfigFromQueryParams();
-
-      // Merge the two config objects
-      const mergedConfig: FlowsInitOptions = {
-        ...configWithoutTranslations,
-        endUserInfo: {
-          ...configWithoutTranslations.endUserInfo,
-          ...endUserInfoFromQueryParams,
-        },
-      };
-
-      const configPromise = appInit(mergedConfig);
-      const translationsPromise = mergeTranslationsOverrides(_translations);
-
-      Promise.all([configPromise, translationsPromise])
+      initConfigAndTranslations(config)
         .then(() => resolve())
         .catch(reject);
     });
   },
 
   mount({
-    flowName = getConfigFromQueryParams().flowName,
-    callbacks,
-    jwt,
-    useModal = false,
-    elementId = '',
-  }) {
+          flowName = getConfigFromQueryParams().flowName,
+          callbacks,
+          jwt,
+          useModal = false,
+          elementId = '',
+        }) {
     const hostElement = useModal
       ? document.querySelector('body')
       : document.getElementById(elementId);
 
-    if (hostElement) {
-      hostElement.innerHTML = `
-        <div class="loader-container" id="blrn-loader">
-            <div class="loader"></div>
-        </div>
-      `;
-    } else {
+    if (!hostElement) {
       const message = useModal ? 'body' : `with id ${elementId}`;
 
-      console.error(`BallerineSDK: Could not find element ${message}`);
+      throw new Error(`BallerineSDK: Could not find element ${message}`);
```

Comment:
I like not having to use `else` and flipping the condition on `hostElement` but throwing an error here is not something we want. If you'd like to try and abstract these specific lines and then using an early return it could work instead of throwing.